src/plugins/Miner.cpp

```diff
@@ -18,13 +18,14 @@ const int mule_energy_cost = 50;
 
 void SecureMineralsIncome(Builder* builder_) {
     auto town_halls = gAPI->observer().GetUnits(sc2::IsTownHall());
+    auto numTotalWorkers = gAPI->observer().CountUnitType(gHub->GetCurrentWorkerType());
 
     for (const auto& i : town_halls()) {
-        if (i->assigned_harvesters >= i->ideal_harvesters)
-            continue;
-
-        if (!i->orders.empty())
+        if (!i->orders.empty() || i->build_progress != 1.0f)
             continue;
+        
+        if (i->assigned_harvesters >= i->ideal_harvesters + 1 || numTotalWorkers > 80)
+            continue;  // NOTE (impulsecloud): one extra, never >80 (pros stop at 77-ish?)
```

Comment:
This value should be set inside a strategy. I I not mistaken, typical zerg has more drones.

Reply:
Added get/set in Hub (since Strat cannot talk to Miner directly). Other option is to put a pointer to Miner in Hub, if you want the member variable in Miner and have Strategy set it.

Reply:
See the comment above.

Reply:
I've removed max_workers limit here, until (#54 "Strategy Settings for plugins") can be done.